longevity_tombstone_gc_test.py

```diff
@@ -63,72 +73,110 @@ def test_switch_tombstone_gc_modes(self):
-        alter_gc_mode_immediate_time = datetime.datetime.now()  # from this point on, all compacted tombstones are GCed.
-        self.log.info('Wait for s-b load to finish')
-        for stress in stress_queue:
-            self.verify_stress_thread(cs_thread_pool=stress)
-        self.log.info('Wait a duration of propagation_delay_in_seconds')
-        time.sleep(self.propagation_delay)
+        _wait_a_duration_of_propagation_delay()
+        _run_and_verify_stress()
+        _wait_a_duration_of_ttl_and_propagation_delay()
         self.db_node.run_nodetool(f"flush -- {self.keyspace}")
         self.log.info('Run a major compaction for user-table on node')
         self.db_node.run_nodetool("compact", args=f"{self.keyspace} {self.table}")
         self.wait_no_compactions_running()
         self.log.info('Verify no compacted tombstones in sstables')
-        #  compaction_gc_delta_minutes = the time range where any compacted tombstone is GCed.
-        compaction_gc_delta_minutes = (datetime.datetime.now() - alter_gc_mode_immediate_time).seconds // 60
-        sstables = sstable_utils.get_sstables(from_minutes_ago=compaction_gc_delta_minutes)
+        sstables = sstable_utils.get_sstables()  # from_minutes_ago=compaction_gc_delta_minutes)
         self.log.debug('Starting sstabledump to verify correctness of tombstones for %s sstables',
                        len(sstables))
         for sstable in sstables:
-            tombstone_deletion_info = sstable_utils.get_compacted_tombstone_deletion_info(sstable=sstable)
+            tombstone_deletion_info = sstable_utils.get_ttl_expired_tombstone_deletion_info(sstable=sstable)
             assert not tombstone_deletion_info, f"Found unexpected existing tombstones: {tombstone_deletion_info} for sstable: {sstable}"
+
+        # Testing gc mode 'disabled'
+        self._drop_and_recreate_keyspace()
+        self.log.info("change gc-grace-seconds to 240 and tombstone-gc mode to 'disabled'")
+        with self.db_cluster.cql_connection_patient(node=self.db_node) as session:
+            query = "ALTER TABLE scylla_bench.test with gc_grace_seconds = 240 " \
+                    f"and tombstone_gc = {{'mode': 'disabled', 'propagation_delay_in_seconds':'{self.propagation_delay}'}};"
+            session.execute(query)
+        _wait_a_duration_of_propagation_delay()
+        _run_and_verify_stress()
+        _wait_a_duration_of_ttl_and_propagation_delay()
+        self.db_node.run_nodetool(f"flush -- {self.keyspace}")
+
+        self.log.info('Count the initial number of tombstones')
+        tombstone_num_pre_repair = sstable_utils.count_ttl_expired_tombstones()
+        self._run_repair_and_major_compaction()
+        tombstone_num_post_repair = sstable_utils.count_ttl_expired_tombstones()
+        assert tombstone_num_post_repair >= tombstone_num_pre_repair, \
+            f"Found unexpected fewer tombstones: {tombstone_num_post_repair} / {tombstone_num_pre_repair}"
```

Comment:
Why do we care we have less tombstones, isnt it better?

Reply:
the gc-mode is "disabled" so tombstones must not be deleted.